File: app/src/main/java/com/scatl/uestcbbs/module/search/presenter/SearchPresenter.java
package com.scatl.uestcbbs.module.search.presenter;

import android.app.AlertDialog;
import android.content.Context;

import com.scatl.uestcbbs.api.ApiConstant;
import com.scatl.uestcbbs.base.BasePresenter;
import com.scatl.uestcbbs.entity.SearchPostBean;
import com.scatl.uestcbbs.entity.SearchUserBean;
import com.scatl.uestcbbs.helper.ExceptionHelper;
import com.scatl.uestcbbs.helper.rxhelper.Observer;
import com.scatl.uestcbbs.helper.rxhelper.SubscriptionManager;
import com.scatl.uestcbbs.module.search.model.SearchModel;
import com.scatl.uestcbbs.module.search.view.SearchView;
import com.scatl.uestcbbs.util.SharePrefUtil;

import io.reactivex.disposables.Disposable;

public class SearchPresenter extends BasePresenter<SearchView> {

    private SearchModel searchModel = new SearchModel();

    public void searchUser(int page, int pageSize, String keyword, Context context) {
        searchModel.searchUser(page, pageSize, 0, keyword,
                SharePrefUtil.getToken(context),
                SharePrefUtil.getSecret(context),
                new Observer<SearchUserBean>() {
                    @Override
                    public void OnSuccess(SearchUserBean searchUserBean) {
                        if (searchUserBean.rs == ApiConstant.Code.SUCCESS_CODE) {
                            view.onSearchUserSuccess(searchUserBean);
                        }
                        if (searchUserBean.rs == ApiConstant.Code.ERROR_CODE) {
                            view.onSearchUserError(searchUserBean.head.errInfo);
                        }
                    }

                    @Override
                    public void onError(ExceptionHelper.ResponseThrowable e) {
                        view.onSearchUserError(e.message);
                    }

                    @Override
                    public void OnCompleted() {

                    }

                    @Override
                    public void OnDisposable(Disposable d) {
                        SubscriptionManager.getInstance().add(d);
                    }
                });
    }

    public void searchPost(int page, int pageSize, String keyword, Context context) {
        searchModel.searchPost(page, pageSize, keyword,
                SharePrefUtil.getToken(context),
                SharePrefUtil.getSecret(context),
                new Observer<SearchPostBean>() {
                    @Override
                    public void OnSuccess(SearchPostBean searchPostBean) {
                        if (searchPostBean.rs == ApiConstant.Code.SUCCESS_CODE) {
                            view.onSearchPostSuccess(searchPostBean);
                        }
                        if (searchPostBean.rs == ApiConstant.Code.ERROR_CODE) {
                            view.onSearchPostError(searchPostBean.head.errInfo);
                        }
                    }

                    @Override
                    public void onError(ExceptionHelper.ResponseThrowable e) {
                        view.onSearchPostError(e.message);
                    }

                    @Override
                    public void OnCompleted() {

                    }

                    @Override
                    public void OnDisposable(Disposable d) {
                        SubscriptionManager.getInstance().add(d);
                    }
                });
    }

}

File: app/src/main/java/com/scatl/uestcbbs/module/search/presenter/SearchPresenter.kt
package com.scatl.uestcbbs.module.search.presenter

import com.scatl.uestcbbs.api.ApiConstant
import com.scatl.uestcbbs.base.BaseVBPresenter
import com.scatl.uestcbbs.entity.SearchPostBean
import com.scatl.uestcbbs.entity.SearchUserBean
import com.scatl.uestcbbs.http.Observer
import com.scatl.uestcbbs.module.search.model.SearchModel
import com.scatl.uestcbbs.module.search.view.SearchView
import com.scatl.uestcbbs.util.subscribeEx

/**
 * Created by sca_tl at 2023/2/8 10:22
 */
class SearchPresenter: BaseVBPresenter<SearchView>() {

    private val searchModel = SearchModel()

    fun searchUser(page: Int, pageSize: Int, keyword: String?) {
        searchModel
            .searchUser(page, pageSize, 0, keyword)
            .subscribeEx(Observer<SearchUserBean>().observer {
                onSuccess {
                    if (it.rs == ApiConstant.Code.SUCCESS_CODE) {
                        mView?.onSearchUserSuccess(it)
                    } else if (it.rs == ApiConstant.Code.ERROR_CODE) {
                        mView?.onSearchUserError(it.head.errInfo)
                    }
                }

                onError {
                    mView?.onSearchUserError(it.message)
                }

                onSubscribe {
                    mCompositeDisposable?.add(it)
                }
            })
    }

    fun searchPost(page: Int, pageSize: Int, keyword: String?) {
        searchModel
            .searchPost(page, pageSize, keyword)
            .subscribeEx(Observer<SearchPostBean>().observer {
                onSuccess {
                    if (it.rs == ApiConstant.Code.SUCCESS_CODE) {
                        mView?.onSearchPostSuccess(it)
                    } else if (it.rs == ApiConstant.Code.ERROR_CODE) {
                        mView?.onSearchPostError(it.head.errInfo)
                    }
                }

                onError {
                    mView?.onSearchPostError(it.message)
                }

                onSubscribe {
                    mCompositeDisposable?.add(it)
                }
            })
    }
}